src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+@RequiredArgsConstructor
+public class ProductController {
+    private final ProductMapper productMapper;
+    private final ProductService productService;
+    private final SortUtil sortUtil;
+
+    @GetMapping
+    @ApiOperation(value = "Get all products")
+    public List<ProductResponseDto> findAll(
+            @RequestParam(defaultValue = "20")
+            @ApiParam(value = "default value is 20") Integer count,
+            @RequestParam(defaultValue = "0")
+            @ApiParam(value = "default value is 0") Integer page,
+            @RequestParam(defaultValue = "id")
+            @ApiParam(value = "default value is id") String sortBy) {
+        Sort sort = sortUtil.createSort(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest).stream()
+                .map(productMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/by-price")
+    @ApiOperation(value = "Get all products with a price between two values")
+    public List<ProductResponseDto> findAllWherePriceBetween(
+            @RequestParam BigDecimal from,
+            @RequestParam BigDecimal to,
+            @RequestParam(defaultValue = "20")
+            @ApiParam(value = "default value is 20") Integer count,
+            @RequestParam(defaultValue = "0")
+            @ApiParam(value = "default value is 0") Integer page,
+            @RequestParam(defaultValue = "id")
+            @ApiParam(value = "default value is id") String sortBy) {
+        Sort sort = sortUtil.createSort(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAllByPriceBetween(from, to, pageRequest).stream()
+                .map(productMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    @ApiOperation(value = "Create a new product")
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        return productMapper.toResponseDto(productService
+                .create(productMapper.toModel(requestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toResponseDto(productService.getById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Delete product by id")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "Update the product by id")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto requestDto) {
```

Comment:
Add description for request body https://stackoverflow.com/questions/64645528/java-spring-boot-openapi-3-how-to-add-description-for-requestbody